client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/Allocation.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect } from 'react';
+import {
+  Typography,
+  InlineSpinner,
+  Box,
+  useTranslation,
+  TableProvider,
+  createTableStore,
+  createQueryParamsStore,
+  ModalRow,
+  ModalLabel,
+  Grid,
+  BasicTextInput,
+} from '@openmsupply-client/common';
+import { OutboundLineEditTable } from './OutboundLineEditTable';
+import { AutoAllocate } from './AutoAllocate';
+import { useOutbound, OutboundLineEditData } from '../../api';
+import { DraftItem } from '../../..';
+import { CurrencyRowFragment } from '@openmsupply-client/system';
+import {
+  useAllocationContext,
+  AllocationStrategy,
+} from './allocation/useAllocationContext';
+import { sumAvailableQuantity } from './allocation/utils';
+
+interface AllocationProps {
+  itemData: OutboundLineEditData;
+  allowPlaceholder: boolean;
+  scannedBatch?: string;
+}
+
+export const Allocation = ({
+  itemData,
+  allowPlaceholder,
+  scannedBatch,
+}: AllocationProps) => {
+  const { initialise, initialisedForItemId } = useAllocationContext(
+    ({ initialise, initialisedForItemId }) => ({
+      initialise,
+      initialisedForItemId,
+    })
+  );
+
+  useEffect(() => {
+    initialise(
```

Comment:
Wrapper component... one time initialisation of the allocation context per item

Reply:
This one confused me, Allocation will 'refresh' because loading will render while fetching. It makes sense once you look at it in more detail, but I still think we can improve on that at some point, almost feel like when item is fetched there is a bunch of things to be done like this, and kind of feels like it should be an await.
This is still readable, i'll just put a comment about this useEffect

Reply:
d3dd097cec